site/source/docs/debugging/Sanitizers.rst

```diff
@@ -0,0 +1,99 @@
+.. _Sanitizers:
+
+=========================
+Debugging with Sanitizers
+=========================
+
+Undefined Behaviour Sanitizer
+=============================
+
+Clang's `undefined behavior sanitizer`__ (UBSan) is now available for use with
+Emscripten. This makes it much easier to catch bugs in your code.
+
+__ https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+
+To use UBSan, simply pass ``-fsanitize=undefined`` to ``emcc`` or ``em++``.
+
+Catching Null Dereference
+-------------------------
+
+With Emscripten, deferencing a null pointer does not immediately cause a
+segmentation fault, unlike traditional platforms. Instead, Emscripten checks
+a magic cookie stored at address 0 at the end of the program execution.
```

Comment:
It's worth mentioning SAFE_HEAP mode here (a special emscripten mode we added to detect out of bounds and other simple memory errors) - that would detect a null deref immediately. But it adds a bunch of overhead - I'm curious how it compares to ubsan?